vpr/src/place/initial_placement.cpp

```diff
@@ -333,17 +514,23 @@ static void place_macro(int macros_max_num_tries, t_pl_macro pl_macro, enum e_pa
             pr.add_to_part_region(reg);
         }
 
-        // Try to place the macro randomly for the max number of random tries
-        for (itry = 0; itry < macros_max_num_tries && macro_placed == false; itry++) {
-            macro_placed = try_random_placement(pl_macro, pr, block_type, pad_loc_type);
+        //if the block type has failed in previous iteration, we need to start place densly to be able to find a legal initial placement solution
+        if (blk_types_empty_locs_in_grid[block_type->index].size() != 0) {
+            macro_placed = try_dense_placement(pl_macro, pr, block_type, pad_loc_type);
+        }
 
-            // Try to place the macro
-            if (macro_placed) {
-                break;
-            }
+        if (!macro_placed) {
+            // Try to place the macro randomly for the max number of random tries
+            for (itry = 0; itry < macros_max_num_tries && macro_placed == false; itry++) {
+                macro_placed = try_random_placement(pl_macro, pr, block_type, pad_loc_type);
 
-        } // Finished all tries
+                // Try to place the macro
+                if (macro_placed) {
```

Comment:
Don't think you need this break --> for loop already will break out after it is placed (and that's more clear than a break).